File: merge-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = ""
        max_len = max(len(word1), len(word2))
        for i in range(max_len):
            if i < len(word1):
                res += word1[i]
            if i < len(word2):
                res += word2[i]
        return res

File: array/merge-alternately.go
func mergeAlternately(word1 string, word2 string) string {
	res := ""
	w1L := len(word1)
	w2L := len(word2)
	maxLen := w1L
	if w2L > w1L {
		maxLen = w2L
	}
	for i := range maxLen {
		if i < w1L {
			res += string(word1[i])
		}
		if i < w2L {
			res += string(word2[i])
		}
	}
	return res
}